test/extended/machines/cluster.go

```diff
@@ -100,4 +119,293 @@ var _ = g.Describe("[sig-node] Managed cluster", func() {
+	}
+	return linesInPodLogs, nil
+}
+
+func createTestBed(ctx context.Context, oc *exutil.CLI) {
+	err := callProject(ctx, oc, true)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	err = callServiceAccount(ctx, oc, true)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	err = callRBAC(ctx, oc, true)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	err = exutil.WaitForServiceAccountWithSecret(
+		oc.AdminKubeClient().CoreV1().ServiceAccounts(namespace),
+		serviceAccountName)
+	o.Expect(err).NotTo(o.HaveOccurred())
+}
+
+func deleteTestBed(ctx context.Context, oc *exutil.CLI) {
+	err := callRBAC(ctx, oc, false)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	err = callServiceAccount(ctx, oc, false)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	err = callProject(ctx, oc, false)
+	o.Expect(err).NotTo(o.HaveOccurred())
+}
+
+func callRBAC(ctx context.Context, oc *exutil.CLI, create bool) error {
+	obj := &authv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceAccountName,
+			Namespace: namespace,
+		},
+		RoleRef: corev1.ObjectReference{
+			Kind: "ClusterRole",
+			Name: "system:openshift:scc:privileged",
+		},
+		Subjects: []corev1.ObjectReference{
+			{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      serviceAccountName,
+				Namespace: namespace,
+			},
+		},
+	}
+
+	client := oc.AdminAuthorizationClient().AuthorizationV1().ClusterRoleBindings()
+	var err error
+	if create {
+		_, err = client.Create(ctx, obj, metav1.CreateOptions{})
+	} else {
+		err = client.Delete(ctx, obj.Name, metav1.DeleteOptions{})
+	}
+
+	if apierrors.IsAlreadyExists(err) || apierrors.IsNotFound(err) {
+		return nil
+	}
+	return err
+}
+
+func callServiceAccount(ctx context.Context, oc *exutil.CLI, create bool) error {
+	obj := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceAccountName,
+			Namespace: namespace,
+		},
+	}
+
+	client := oc.AdminKubeClient().CoreV1().ServiceAccounts(namespace)
+	var err error
+	if create {
+		_, err = client.Create(ctx, obj, metav1.CreateOptions{})
+	} else {
+		err = client.Delete(ctx, obj.Name, metav1.DeleteOptions{})
+	}
+
+	if apierrors.IsAlreadyExists(err) || apierrors.IsNotFound(err) {
+		return nil
+	}
+	return err
+}
+
+func callProject(ctx context.Context, oc *exutil.CLI, create bool) error {
+	obj := &projv1.Project{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+			Labels: map[string]string{
+				"pod-security.kubernetes.io/audit":   "privileged",
+				"pod-security.kubernetes.io/enforce": "privileged",
+				"pod-security.kubernetes.io/warn":    "privileged",
+			},
+		},
+	}
+
+	client := oc.AsAdmin().ProjectClient().ProjectV1().Projects()
+	var err error
+	if create {
+		_, err = client.Create(ctx, obj, metav1.CreateOptions{})
+	} else {
+		err = client.Delete(ctx, obj.Name, metav1.DeleteOptions{})
+	}
+
+	if apierrors.IsAlreadyExists(err) || apierrors.IsNotFound(err) {
+		return nil
+	}
```

Comment:
I'm a little uncomfortable with a bool parameter toggling between create and delete as there isn't much reuse between the two paths.

Not that I imagine this specific test running on MicroShift, but with MicroShift we now have a variant that doesn't serve the project API at all. I'd expect a "create project" call to return 404 in that case.

It might be convenient to use https://github.com/openshift/origin/blob/626c3ff421d2c9f169765bb4487187cca3a627cf/test/extended/util/client.go#L134 and create only the ServiceAccount and RoleBinding fixtures.

Reply:
Right, its not looking great. I think its best to remove this flag, create just the namespace-scoped objects and rely on namespace finalization instead